content/blog/2025/fluid-type.md

```diff
@@ -0,0 +1,404 @@
+
+As a web surfer,
+I would like my default font size
+to respond to those changes.
+As a web author,
+I would like my site to be responsive as well.
+We design layouts that respond smoothly
+from the smallest screen to the largest --
+it seems absurd to keep the base font size
+locked in place through that entire responsive process.
+
+So _I want responsive typography_,
+and browsers (as a rule)
+don't make that available in user settings.
+There are some browsers (like Safari)
+that accept arbitrary CSS preferences,
+but that's rare --
+and only useful to people who write CSS.
+
+There are a few ways I could imagine
+approaching this.
+Maybe we just add a slight scale factor
+to the user default:
+
+```css
+html {
+  font-size: calc(1em + 1vw);
+}
+```
+
+But that will always be larger than the user asked for.
+So we could adjust it down slightly,
+and provide clamps:
+
+```css
+html {
+  font-size: calc(1em, 0.9em + 1vw, 1.5em);
+}
+```
+
+This looks a lot like the common approach in structure,
+but it has some distinct differences.
+We're not trying to achieve a specific font size
+(or range of sizes)
+by assuming the user has a `16px` default.
+There's no _calculation witchcraft_ aimed
+at some target value in our designer mind,
+and the clamps are not related to my desired range of outcomes.
+Instead we're adding a slight responsiveness to the user value,
+and using the clamps to keep our fluid value
+within range of the user's intent.
+
+Relative units like `em` and `rem`
+are excellent for _internal relationships_ --
+keeping the scale consistent between
+body text and headings for example.
+But things always go wrong
+when we try to treat `em` as an alias for `px`,
+with mental conversions based on their assumed default relationship.
+At that point we're keeping _zoom_,
+but discarding everything else about the user preference.
+
+Any time you start doing mental math with `16px == 1em`,
+stop yourself and ask if that math holds up
+over a whole range of user preferences.
+
+(It doesn't. Don't do it.)
+
+## Doing what the browser won't
+
+I also have an issue
+where my web-surfing preference for large text
+does not apply equally to all sites.
+When I'm reading a blog post,
+I just want a nice big view of the text.
+When I'm scanning bank transactions,
+I'm happy for something a bit more compact.
+I think browsers like Arc are on the right track
+with per-site style adjustments.
+
+We could take my responsive-default approach even farther
+by providing site-specific controls
+as part of our UI.
+We still want to make sure the defaults are reasonable,
+based on the global preferences,
+and only provide the UI as a progressive enhancement.
+To quote accessibility expert Kate Kalcevich:
+
+{{ quotes.blockquote({
+  'text': "The real advance in accessibility is providing options and adapting to user preferences.",
+  'name': "Kate Kalcevich"
+}) }}
```

Comment:
Can we link to a source for this quote? Not a big deal.

Reply:
It was a talk that I saw in person, so not sure if there's a link

Reply:
Found it, link added